pennylane/fourier/visualization.py

```diff
@@ -0,0 +1,717 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains visualization functions for Fourier series and coefficients."""
+import warnings
+from itertools import product
+import numpy as np
+
+# Matplotlib is not a hard requirement for PennyLane in general, but it *is*
+# a hard requirement for everything in this module.
+try:
```

Comment:
I've removed the `from .visualization import...` from the `__init__.py` so that the coefficients module can be imported without the user receiving warnings about `matplotlib`. Let me know if this is not what you had in mind re. decoupling the import.